File: frontend/src/shared/models/FoodDto.ts
export type FoodType = {
	calories: number
	carbs: number
	fat: number
	id: number
	name: string
	protein: number
};

export class FoodDto {
	public calories: number = 0;
	public carbs: number = 0;
	public fat: number = 0;
	public id: number = 0;
	public name: string = '';
	public protein: number = 0;

	constructor(base: FoodType) {
		this.calories = base.calories;
		this.carbs = base.carbs;
		this.fat = base.fat;
		this.id = base.id;
		this.name = base.name;
		this.protein = base.protein;
	}
}

File: backend/Coacher.Backend.Contracts/Dto/FoodDto.cs
namespace Coacher.Backend.Contracts.Dto;

public class FoodDto
{
    public Guid? Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public double Calories { get; set; } = 0;
    public double Protein { get; set; } = 0;
    public double Carbs { get; set; } = 0;
    public double Fat { get; set; } = 0;
}
